### DISCUSSION TITLE
Will this plugin support the new Corsair Link hub and QX fans?

### DISCUSSION THREAD

**Discussion by IssacI19** [EvanMulawski/FanControl.CorsairLink#86] [General]
I just wanted to ask if at any point in time the CorsairLink FanControl plugin will work for the new Corsair Link hub, AIO, and QX fans. Because at one point I'm thinking of upgrading to that to free some cable clutter. Thank you

**Comment by EvanMulawski** [comment]
Assuming you're referring to the new "iCUE LINK" product line, developers will need annotated USB captures from someone who owns one of those devices before work can begin.

**Comment by vladimirgluten** [comment]
How would someone do that USB capture?  I just got 7 QX Fans and the iCUE Link hub and would LOVE to have FanControl control those fans, as the corsair software is....not very good.

**Reply by EvanMulawski** [reply]
Thanks!

1. Install the latest version of Wireshark and install USBPcap when prompted during the installation process. 
2. Launch Wireshark as administrator and start a USBPcap capture.
3. Launch iCUE and wait one minute to capture any initial communication packets.
4. In iCUE, change one fan and note the time when you are changing it. In Wireshark, right click a packet as close to that time as possible and add a comment to it noting what you changed, e.g. "set fan 3 fixed duty 99%". (For Fan Control, I only need to know how to set a fan to fixed duty/percent - changing the fan mode to fixed duty/percent from a different mode would be helpful to see.)
5. Repeat Step 4 for all fans.
6. Save the capture (should be a pcapng file), ZIP it, and upload it here.

Let me know if you need any additional help or information.

**Reply by MaxManHH** [reply]
I can help with providing information too if needed ( I got the QX120 Fans and the H150i RGB with the System-Hub) 
I would love to have Integration of those

**Reply by XeTrinity** [reply]
Hello I also have these fans and would love for them to be added, so i did what you asked and have provided the file below.
I had to upload it to my google drive because its over the 25mb limit :/
If i did anything wrong please contact me ill be glad to redo it :)

[Corsair Pcap zip](https://drive.google.com/file/d/1h_T76PAxdHx1JCX1_41UuBw1xYr-c03U/view?usp=sharing)

**Reply by jhiggins** [reply]
I’ve also just set up a 170 AIO and fans plus a 120 fan. I’m not sure if fan sizes have different USB packets, but I can also run a pcap if it helps.

**Reply by jhiggins** [reply]
Here is a dump from my system as well, 

https://www.dropbox.com/scl/fi/1jf89z9er96v3strthyi4/H170-10-fans.pcapng?rlkey=9nedk7qpnbblr4kb0qlyz8nh0&dl=0

H170I + 10 fans.  Fan 4 is a QX120 and the rest are QX140s.  I don't really know much about USB formats, but I  annotated the second where I started the change.  I use 95% and Off as my cycles and changed it back to the fan curve I have configured in ICUE in between.

Hopefully this is useful.

**Reply by jhiggins** [reply]
So digging into this a little, here is a filter for my file where you can see packets that seem to be where I was setting fans to 95%.  You can see the 0x5F (95) progressing from position to position as I walk up the list of fans in the usbhid field.

```
!(usbhid.data contains 00:01:06:00:3a:04) && !(usbhid.data contains 00:01:07:00) && usb.dst == "5.3.4" && usbhid.data contains 5F
```

They happen in the right window of times where I was setting fans to 95%.  This could also be a complete red herring

**Reply by jhiggins** [reply]
After digging and some live captures I think the fan speed packet is very similar to what's in the current code for the Commander Core.  

A prefix of 0x000106012F0000000700
Number of devices (0x0B, 10 fans + AIO)
Then a list of channel + control type + speed 

Control type appears to be 00 for percentages, 01 for fixed RPM with a 16 bit little endian speed, 
Examples: 0x01005F00 for channel 1 at 95% and 0x01015802 for channel 1 at 600rpm

I can help dig some more if it's useful.

**Reply by a-zndr** [reply]
[2023-11-13-1-pcap.txt](https://github.com/EvanMulawski/FanControl.CorsairLink/files/13340203/2023-11-13-1-pcap.txt)

here's a quick pcap of me changing a bunch of settings on my fans+ AIO 
<img width="1539" alt="Screenshot 2023-11-13 112940" src="https://github.com/EvanMulawski/FanControl.CorsairLink/assets/14003031/483a946f-f56a-4f8c-b622-99bf6ff05794">

**Reply by EvanMulawski** [reply]
@ScoPeZs What model liquid cooler?

@ScoPeZs @jhiggins @a-zndr Could you let me know how everything is connected together?

**Reply by jhiggins** [reply]
In terms of cable order?

**Reply by jhiggins** [reply]
I have two basic daisy chains. One going off of a y cable and another long one from the AIO

```
Link controller
| - second controller Port
    | y-cable 1 - Fan 4
    | y-cable 2 
      - Fan 3
      - Fan 1
      - Fan 2
|  first controller port
  | AIO Splitter
    | fan 8
    | fan 9
    | fan 10
    | fan 7
    | fan 5
    | fan 6
```

**Reply by XeTrinity** [reply]
I have the Corsair iCUE LINK H150i 360mm.
My fans are connected like so. Single 120 to Tri connected 120s into mini split hub on the AIO from that to the ICUE Link hub 2nd tri connected 120s connected to the same ICUE Link hub.
![Untitled](https://github.com/EvanMulawski/FanControl.CorsairLink/assets/16966844/02be5b0a-b70d-4dee-a34b-ce02d975d44f)

**Reply by a-zndr** [reply]
I have an almost identical setup to Scopezs 
![Screenshot 2023-11-19 155214](https://github.com/EvanMulawski/FanControl.CorsairLink/assets/14003031/dcd24bbd-c056-409c-946b-1821f804ee50)


2 140's on one line to the hub, 3 120s daisy chained (1 free standing 2 on the AIO) and the pump at the end

**Comment by EvanMulawski** [comment]
@jhiggins Could you provide a brief capture during iCUE startup when you get a chance? i.e. Quit iCUE, start a capture, then launch iCUE. Just need one minute after starting iCUE. Thanks!

**Reply by jhiggins** [reply]
I started iCue almost immediately after starting the capture.

[icue-startup.pcapng.gz](https://github.com/EvanMulawski/FanControl.CorsairLink/files/13360843/icue-startup.pcapng.gz)

**Comment by EvanMulawski** [comment]
Closing in favor of #111
